src/common/transformations/src/transformations/common_optimizations/nop_elimination.cpp

```diff
@@ -653,6 +657,8 @@ pass::EliminateConcatStridedSlice::EliminateConcatStridedSlice() {
                 new_concat_in_nodes.push_back(concat_input_node);
             }
         }
+        if (new_concat_in_nodes.size() == concat_inputs.size())
+            return false;
```

Comment:
are you sure about `return false` here?
it looks like, the graph may have changed earlier
<img width="393" alt="{0FF8C86F-121E-49DB-8D64-F708CDFA1027}" src="https://github.com/user-attachments/assets/9f0fadf7-5021-47ab-b6eb-3e96362fe98a" />
we usually `return true` if the graph structure has been changed.